### ISSUE TITLE
Bad ssl certificat on a feed rise an Serveur Error 500

### ISSUE THREAD

**Issue by pieplu** [OleVik/grav-plugin-twigfeeds#18]
If an twig_feeds source have an https link who are a bad certificat ssl, we have the error on the front:
It will be better to don't rise an error for all th website if we are not on a debug mode

```
0 - PicoFeed\Client\InvalidCertificateException: Invalid SSL certificate caused by CURL error number 60 in [hidden-path]/grav/user/plugins/twigfeeds/vendor/fguillot/picofeed/lib/PicoFeed/Client/Curl.php:389
Stack trace:
#0 [hidden-path]/grav/user/plugins/twigfeeds/vendor/fguillot/picofeed/lib/PicoFeed/Client/Curl.php(281): PicoFeed\Client\Curl-&gt;handleError(60)
#1 [hidden-path]/grav/user/plugins/twigfeeds/vendor/fguillot/picofeed/lib/PicoFeed/Client/Curl.php(297): PicoFeed\Client\Curl-&gt;executeContext()
#2 [hidden-path]/grav/user/plugins/twigfeeds/vendor/fguillot/picofeed/lib/PicoFeed/Client/Client.php(218): PicoFeed\Client\Curl-&gt;doRequest()
#3 [hidden-path]/grav/user/plugins/twigfeeds/vendor/fguillot/picofeed/lib/PicoFeed/Reader/Reader.php(53): PicoFeed\Client\Client-&gt;execute('https://url....')
#4 [hidden-path]/grav/user/plugins/twigfeeds/Parser.php(68): PicoFeed\Reader\Reader-&gt;download('https://url....')
#5 [hidden-path]/grav/user/plugins/twigfeeds/twigfeeds.php(214): TwigFeeds\Parser-&gt;parseFeed(Array, '[hidden-path]...')
#6 [internal function]: Grav\Plugin\TwigFeedsPlugin-&gt;outputFeeds(Object(RocketTheme\Toolbox\Event\Event), 'onTwigSiteVaria...', Object(RocketTheme\Toolbox\Event\EventDispatcher))
#7 [hidden-path]/grav/vendor/symfony/event-dispatcher/EventDispatcher.php(184): call_user_func(Array, Object(RocketTheme\Toolbox\Event\Event), 'onTwigSiteVaria...', Object(RocketTheme\Toolbox\Event\EventDispatcher))
#8 [hidden-path]/grav/vendor/symfony/event-dispatcher/EventDispatcher.php(46): Symfony\Component\EventDispatcher\EventDispatcher-&gt;doDispatch(Array, 'onTwigSiteVaria...', Object(RocketTheme\Toolbox\Event\Event))
#9 [hidden-path]/grav/vendor/rockettheme/toolbox/Event/src/EventDispatcher.php(23): Symfony\Component\EventDispatcher\EventDispatcher-&gt;dispatch('onTwigSiteVaria...', Object(RocketTheme\Toolbox\Event\Event))
#10 [hidden-path]/grav/system/src/Grav/Common/Grav.php(288): RocketTheme\Toolbox\Event\EventDispatcher-&gt;dispatch('onTwigSiteVaria...', Object(RocketTheme\Toolbox\Event\Event))
#11 [hidden-path]/grav/system/src/Grav/Common/Twig/Twig.php(321): Grav\Common\Grav-&gt;fireEvent('onTwigSiteVaria...')
#12 [hidden-path]/grav/system/src/Grav/Common/Service/OutputServiceProvider.php(27): Grav\Common\Twig\Twig-&gt;processSite('html')
#13 [hidden-path]/grav/vendor/pimple/pimple/src/Pimple/Container.php(118): Grav\Common\Service\OutputServiceProvider-&gt;Grav\Common\Service\{closure}(Object(Grav\Common\Grav))
#14 [hidden-path]/grav/system/src/Grav/Common/Processors/RenderProcessor.php(19): Pimple\Container-&gt;offsetGet('output')
#15 [hidden-path]/grav/system/src/Grav/Common/Grav.php(132): Grav\Common\Processors\RenderProcessor-&gt;process()
#16 [hidden-path]/grav/system/src/Grav/Common/Grav.php(379): Grav\Common\Grav-&gt;Grav\Common\{closure}()
#17 [internal function]: Grav\Common\Grav::Grav\Common\{closure}('render', 'Render', Object(Closure))
#18 [hidden-path]/grav/system/src/Grav/Common/Grav.php(355): call_user_func_array(Object(Closure), Array)
#19 [hidden-path]/grav/system/src/Grav/Common/Grav.php(133): Grav\Common\Grav-&gt;__call('measureTime', Array)
#20 [hidden-path]/grav/index.php(52): Grav\Common\Grav-&gt;process()
#21 {main}
```

**Comment by OleVik** [comment]
The library will throw that error regardless, and suppressing it on that end would be a security risk. Creating a bypass would be inefficient, as it poses a security-risk and site admins need to test connections beforehand.

**Comment by pieplu** [comment]
Thanks to respond me,
I don't say to suppressing the error, but for "production" config (at least, no debug mode or other), display it on log, and return (for exemple) an empty Array. No security-risk, and **no website page down**. Only an empty section on the site.
With the actual behaviour, an ssl error on the certificat of the New-York-time website (or other external feed website), can result on your entire website down. **It's a major stability risk**!

But I can understant your point too. It can be an Grav's architecture issue too.

**Comment by OleVik** [comment]
I will evaluate the possibilities further, you make a fair point in stating that it is a stability-issue that the site itself cannot handle.

**Comment by pieplu** [comment]
Thanks! :)

**Comment by ghost** [comment]
@OleVik @pieplu I am preparing a patch to fix this issue. How much time does it take to be available as an update in GPM ? Thanks !

**Comment by OleVik** [comment]
Only an hour once it's in the master-branch, though I do not blindly accept PRs without a proper code review first, so expect a delay. Any change to code can be manually integrated though.

**Comment by OleVik** [comment]
A fix is available in [v3.3.0-beta.1.1](https://github.com/OleVik/grav-plugin-twigfeeds/releases/tag/v3.3.0-beta.1.1), which will be available on the testing-channel of the GPM soon.

**Comment by ghost** [comment]
@OleVik Thanks my dude. Any idea of when it will be available in stable ?

**Comment by OleVik** [comment]
Sometime this week.

**Comment by OleVik** [comment]
Released v3.3.0, GPM should pick it up to stable channel within an hour.

**Comment by ghost** [comment]
@OleVik Excellents, thank you very much !

Have a great day !
